src/puppetlabs/puppetdb/http/command.clj

```diff
@@ -36,12 +38,34 @@
 
         :else (app req)))))
 
+(defn- blocking-submit-command [do-submit-fn response-pub uuid completion-timeout-ms]
+  (let [response-chan (async/chan)
+        topic (str uuid)
+        _ (async/sub response-pub topic response-chan)
+        timeout-chan (async/timeout completion-timeout-ms)
+        _ (do-submit-fn)
+        timed-out (async/alt!! timeout-chan true
+                               response-chan false)]
+    (async/unsub response-pub topic response-chan)
+    (http/json-response {:uuid uuid, :processed (not timed-out)}
+                        (if timed-out 503 200))))
+
```

Comment:
There should definitely be a `finally` with a call to `unsub`. I'm less certain about needing to close `response-chan`; http://stackoverflow.com/questions/28888340/should-clojure-core-async-channels-be-closed-when-not-used-anymore indicates that it may not be needed. Looking through https://github.com/clojure/core.async/blob/master/src/main/clojure/clojure/core/async/impl/channels.clj, this *may* be true. Of note are `puts` and `takes` members of `ManyToManyChannel`, which are each a linked list of mutexes. Will continue to investigate.